### PR TITLE
Removes yet another xenobio exploit

### PR DESCRIPTION
**Pull request by Dejaku51**
## About The Pull Request
extract do_after was attached to the turf player was on instead of the extract so it didn't cancel when extract was deleted resulting in a lot of pylons if spammed. This PR fixes it.
Also someone could place another pylon while yours was being constructed because it checked for pylons only before the progress bar

## Why It's Good For The Game
Xenobio exploits bad

## Testing Photographs and Procedure
- Try to spam xenobio crystalline pylons
- Progressbars vanish/cancel when one pylon gets created

<details>
<summary>Screenshots&Videos</summary>

Current master behavior:
image
image

This PR behavior:
image
image

</details>

## Changelog
:cl:
fix: Fixed xenobio infinite pylons exploit that could potentially crash the server
/:cl:

### CODE REVIEW

File: code/modules/research/xenobiology/crossbreeding/crystalized.dm [hunk 1]
```diff
@@ -17,7 +17,13 @@
 
 	var/user_turf = get_turf(user)
 
-	if(!do_after(user, 15 SECONDS, user_turf, timed_action_flags = IGNORE_HELD_ITEM))
+	if(!do_after(user, 15 SECONDS, src))
+		return
+
+	// check after in case someone placed a crystal in the meantime (im watching you aramix)
+	var/obj/structure/slime_crystal/C = locate(/obj/structure/slime_crystal) in range(6,get_turf(user))
+	if(C)
```
Comment (PowerfulBacon) [inline code/modules/research/xenobiology/crossbreeding/crystalized.dm:25]: Check if src is qdeleted too

**Review by PowerfulBacon — APPROVED** [review]

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
There isn't spacing between functions and this solution is kind of ugly.
Just do
```
if(locate(/obj/structure/slime_crystal) in range(6,get_turf(user)))
	to_chat(user,"<span class='notice'>You can't build crystals that close to each other!</span>")
```

File: code/modules/research/xenobiology/crossbreeding/crystalized.dm [hunk 2]
```diff
@@ -9,15 +9,21 @@
 /obj/item/slimecross/crystalline/attack_self(mob/user)
 	. = ..()
 
+	// Check before the progress bar so they don't wait for nothing
 	var/obj/structure/slime_crystal/C = locate(/obj/structure/slime_crystal) in range(6,get_turf(user))
```
Comment (PowerfulBacon) [inline code/modules/research/xenobiology/crossbreeding/crystalized.dm]: Unused variable

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/modules/research/xenobiology/crossbreeding/crystalized.dm (hunk 2 above)
Comment (Dejaku51) [reply code/modules/research/xenobiology/crossbreeding/crystalized.dm]: fixed

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**